pkg/pilot/cassandra/v3/pilot.go

```diff
@@ -127,3 +124,68 @@ func (p *Pilot) LivenessCheck() error {
 	_, err := p.nodeTool.Status()
 	return err
 }
+
+func (p *Pilot) writeCassandraYaml(pilot *v1alpha1.Pilot) error {
+	cfg, err := config.NewFromYaml(
+		path.Join(p.Options.CassandraConfigPath, "cassandra.yaml"),
+	)
+	if err != nil {
+		return err
+	}
+	cfg.Set("cluster_name", p.Options.CassandraClusterName)
+	// We unset any pre-configured IP / host addresses so that cassandra is
+	// forced to lookup of its FQDN and corresponding IP address.
+	// This allows cassandra to handle changes of IP address if Pods are restarted.
+	cfg.Unset("listen_address")
+	cfg.Unset("listen_interface")
+	cfg.Unset("broadcast_address")
+	cfg.Unset("rpc_address")
+	// Force the use of GossipingPropertyFileSnitch so that cassandra will query the
+	// rackdc properties file for rack and DC values.
+	cfg.Set("endpoint_snitch", CassSnitch)
+	// Force the use of Kubernetes seed provider
+	// And remove preconfigured seeds so that cassandra can only get seed
+	// information from the Kubernetes service.
+	cfg.Set(
+		"seed_provider",
+		[]map[string]interface{}{{
+			"class_name": CassSeedProvider,
+			"parameters": []interface{}{
+				map[string]interface{}{
+					"seeds": "",
+				},
+			},
+		}},
```

Comment:
🤔 Is this a slice, of maps, of strings to interfaces? (of which one of those elements is a slice of interfaces, containing a map from strings to interfaces?!)

I guess this is just because of how Viper works (due to lack of generics), and how the YAML is structured, but... wow